docs/references/uvh5_memo.tex

```diff
@@ -0,0 +1,509 @@
+italics, \textbf{(3)} a brief description of the data, and \textbf{(4)} the name
+of the corresponding attribute on a UVData object. Note that unlike in other
+formats, names of HDF5 datasets can be quite long, and so in most cases the name
+of the dataset corresponds to the name of the UVData attribute.
+
+Note that string datatypes should be handled with care. See
+Appendix~\ref{appendix:strings} for appropriately defining them for
+interoperability between different HDF5 implementations.
+
+\subsection{Required Parameters}
+\label{sec:req_params}
+\begin{itemize}
+\item \textbf{latitude}: \textit{float} The latitude of the telescope site, in
+  degrees. (\textit{latitude})
+\item \textbf{longitude}: \textit{float} The longitude of the telescope site, in
+  degrees. (\textit{longitude})
+\item \textbf{altitude}: \textit{float} The altitude of the telescope site, in
+  meters. (\textit{altitude})
+\item \textbf{instrument}: \textit{string} The name of the instrument, typically
+  the telescope name. (\textit{instrument})
+\item \textbf{object\_name}: \textit{string} The name of the object tracked by
+  the telescope. For a drift-scan antenna, this is typically
+  ``zenith''. (\textit{object\_name})
+\item \textbf{history}: \textit{string} The history of the data
+  file. (\textit{history})
+\item \textbf{phase\_type}: \textit{string} The phase type of the
+  observation. Should be ``phased'' or ``drift''. Any other value is treated as
+  an unrecognized type. (\textit{phase\_type})
+\item \textbf{Nants\_data}: \textit{int} The number of antennas that data in the
+  file corresponds to. May be smaller than the number of antennas in the
+  array. (\textit{Nants\_data})
+\item \textbf{Nants\_telescope}: \textit{int} The number of antennas in the
+  array. May be larger than the number of antennas with data corresponding to
+  them. (\textit{Nants\_telescope})
+\item \textbf{ant\_1\_array}: \textit{int} An array of the first antenna indices
+  corresponding to baselines present in the data. This is a one-dimensional
+  array of size Nblts. (\textit{ant\_1\_array})
+\item \textbf{ant\_2\_array}: \textit{int} An array of the second antenna indices
+  corresponding to baselines present in the data. This is a one-dimensional
+  array of size Nblts. (\textit{ant\_2\_array})
+\item \textbf{antenna\_names}: \textit{string} An array of the names of antennas
+  present in the array. This is a one-dimensional array of size
+  Nants\_telescope. Note there must be one entry for every unique antenna in
+  ant\_1\_array and ant\_2\_array, but there may be additional
+  entries. (\textit{antenna\_names})
+\item \textbf{baseline\_array}: \textit{int} An array of baseline indices
+  corresponding to the data. This is a one-dimensional array of size Nblts. The
+  baseline index is calculated as:
+  \[
+    \mathtt{baseline} = 2048*(\mathtt{ant2}+1)+(\mathtt{ant1}+1) + 2^{16}.
+  \]
+  For current implementations, 32-bit integers (an \verb+int+ type in C) are
```

Comment:
should "type in C" be "type of C"?

Also, reading through this I realized that the baseline_array and ant_1_array + ant_2_array is really providing redundant info. That's perfectly fine, but maybe the baseline_array should be optional (in the file format, not on the object)?

Reply:
good point re: the baseline array. I've noticed that whenever we're doing our antenna selection with, e.g., `get_data`, the ant_1 and ant_2 arrays are what's used anyway, so making this array optional is not a bad idea.